src/TestExplorer/TestRunner.ts

```diff
@@ -233,105 +188,122 @@ export class TestRunner {
-                    return null;
+    /** Run test session without attaching to a debugger */
+    async runSession(
+        token: vscode.CancellationToken,
+        testKind: TestKind,
+        runState: TestRunnerTestRunState
+    ) {
+        // Run swift-testing first, then XCTest.
+        // swift-testing being parallel by default should help these run faster.
+        if (this.testArgs.hasSwiftTestingTests) {
+            const fifoPipePath =
+                process.platform === "win32"
+                    ? `\\\\.\\pipe\\vscodemkfifo-${Date.now()}`
+                    : path.join(os.tmpdir(), `vscodemkfifo-${Date.now()}`);
+
+            await TemporaryFolder.withNamedTemporaryFile(fifoPipePath, async () => {
+                // macOS/Linux require us to create the named pipe before we use it.
+                // Windows just lets us communicate by specifying a pipe path without any ceremony.
+                if (process.platform !== "win32") {
+                    await execFile("mkfifo", [fifoPipePath], undefined, this.folderContext);
                 }
-                return testBuildConfig;
-            } else {
-                const testBuildConfig = createTestConfiguration(this.folderContext, true);
+
+                const testBuildConfig =
+                    await LaunchConfigurations.createLaunchConfigurationForSwiftTesting(
+                        this.testArgs.swiftTestArgs,
+                        this.folderContext,
+                        fifoPipePath
+                    );
+
                 if (testBuildConfig === null) {
-                    return null;
+                    return;
                 }
 
-                if (testList.length > 0) {
-                    testBuildConfig.args = ["-XCTest", testList, ...testBuildConfig.args];
+                // Output test from stream
+                const outputStream = new stream.Writable({
+                    write: (chunk, encoding, next) => {
+                        const text = chunk.toString();
+                        this.testRun.appendOutput(text.replace(/\n/g, "\r\n"));
+                        next();
+                    },
+                });
+
+                if (token.isCancellationRequested) {
+                    outputStream.end();
+                    return;
                 }
-                // output test logging to debug console so we can catch it with a tracker
-                testBuildConfig.terminal = "console";
-                return testBuildConfig;
-            }
-        } else {
-            const testBuildConfig = createTestConfiguration(this.folderContext, true);
+
+                // Watch the pipe for JSONL output and parse the events into test explorer updates.
+                // The await simply waits for the watching to be configured.
+                await this.swiftTestOutputParser.watch(fifoPipePath, runState);
+
+                await this.launchTests(
+                    testKind,
```

Comment:
Given parallel test profile is specific to the XCTests (Swift testing already runs in parallel). Perhaps this should not use the parallel test path here.
```suggestion
                    testKind === TestKind.parallel ? TestKind.standard : testKind,
```

Reply:
swift-testing honours parallel/no-parallel options at the command line. The default is reversed from XCTest. Dunno if that affects your suggestion or not.

Reply:
Good idea. Eventually we'll want to offer running swift-testing tests with no-parallel, but for now I think it makes sense to run them as their default.

Reply:
@grynspan this was more about getting swift-testing working when someone chooses parallel. The parallel path at the moment breaks swift-testing. Eventually we should have the parallel/non-parallel setting work for both XCTest and testing. 

Parallel for XCTest doesn't work great in VSCode (result only available at end, error parsing broken prior to Swift 6) so wasn't keen on making it the default. Meanwhile swift-testing parallel works great so would be a shame to not make it the default. Perhaps we should have two settings.